static/app/chartcuterie/performance.tsx

```diff
@@ -23,11 +26,55 @@ function modifyOptionsForSlack(options: Omit<LineChartProps, 'series'>) {
     visualMap: options.options?.visualMap,
   };
 }
+type FunctionRegressionChartData = Omit<
+  EventBreakpointChartData,
+  'chartType' | 'percentileData'
+> & {rawResponse: any};
 
 performanceCharts.push({
   key: ChartType.SLACK_PERFORMANCE_ENDPOINT_REGRESSION,
-  getOption: (data: EventBreakpointChartData) => {
-    const {chartOptions, series} = getBreakpointChartOptionsFromData(data, theme);
+  getOption: (data: EndpointRegressionChartData) => {
+    const param = {
+      ...data,
+      chartType: ChartType.SLACK_PERFORMANCE_ENDPOINT_REGRESSION,
+    };
+
+    const {chartOptions, series} = getBreakpointChartOptionsFromData(param, theme);
+    const transformedSeries = transformToLineSeries({series});
+    const modifiedOptions = modifyOptionsForSlack(chartOptions);
+
+    return {
+      ...modifiedOptions,
+
+      backgroundColor: theme.background,
+      series: transformedSeries,
+      grid: slackChartDefaults.grid,
+      visualMap: modifiedOptions.options?.visualMap,
+    };
+  },
+  ...slackChartSize,
+});
+
+performanceCharts.push({
+  key: ChartType.SLACK_PERFORMANCE_FUNCTION_REGRESSION,
+  getOption: (data: FunctionRegressionChartData) => {
+    const transformed = transformStatsResponse(
+      'profileFunctions',
+      ['p95()'],
+      data.rawResponse
+    );
+
+    const percentileData = {
+      data: transformed,
+    };
+
+    const param = {
+      percentileData: percentileData as any,
```

Comment:
i don't like to type this as any, but couldn't figure out a better way

Reply:
oh I just added a comment below. In ts you can just `|` all the possible types. Can't you?